public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -69,18 +130,58 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   } = useFormContext<RuleFormValues>();
 
   const { queryPreviewData, runQueries, cancelQueries, isPreviewLoading, clearPreviewData } = useAlertQueryRunner();
+  const isSwitchModeEnabled = config.featureToggles.alertingQueryAndExpressionsStepMode ?? false;
 
   const initialState = {
     queries: getValues('queries'),
   };
 
   const [{ queries }, dispatch] = useReducer(queriesAndExpressionsReducer, initialState);
+
+  // data queries only
+  const dataQueries = useMemo(() => {
+    return queries.filter((query) => !isExpressionQuery(query.model));
+  }, [queries]);
+
+  // expression queries only
+  const expressionQueries = useMemo(() => {
+    return queries.filter((query) => isExpressionQuery(query.model));
+  }, [queries]);
+
   const [type, condition, dataSourceName] = watch(['type', 'condition', 'dataSourceName']);
 
   const isGrafanaAlertingType = isGrafanaAlertingRuleByType(type);
   const isRecordingRuleType = isCloudRecordingRuleByType(type);
   const isCloudAlertRuleType = isCloudAlertingRuleByType(type);
 
+  const [isAdvancedMode, setIsAdvancedMode] = useState(isGrafanaAlertingType && isSwitchModeEnabled ? false : true);
+
+  const expressionQueriesList = useMemo(() => {
```

Comment:
What's the difference here between `expressionQueries` and `expressionQueriesList`?

Reply:
I pushed a commit removing the use of this `expressionQueriesList` and using a new type guard for `expressionQueries`